src/sentry/sentry_metrics/indexer/limiters/cardinality.py

```diff
@@ -99,22 +108,31 @@ def check_cardinality_limits(
                 ),
                 16,
             )
-            prefix = _build_quota_key(self.namespace, org_id)
+            prefix = _build_quota_key(message["use_case_id"], org_id)
```

Comment:
`message["use_case_id"]` is likely a string, not an enum. but `_build_quota_key` needs an enum. if this sort of thing isn't caught by mypy we should investigate why

Reply:
Actually, `message["use_case_id"]` is an Enum. `InboundMessage` type shows this

Reply:
that can't be true. this is a JSON payload. if the type annotation on `InboundMessage` says that then the type annotation is wrong

Reply:
oh i see: https://github.com/getsentry/sentry/blob/4bb3321c6e864fcdbeac1d868d5387356f6af13a/src/sentry/sentry_metrics/consumers/indexer/batch.py#L78

I don't think mutating the message here is great, but you are right